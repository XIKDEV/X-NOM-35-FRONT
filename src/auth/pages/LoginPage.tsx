import { LoginPageHooks } from '../hooks';

import { authConstants } from '../../constants/authConstants';

import '../ui/loginPage.css';

export const LoginPage = () => {
  LoginPageHooks();

  return (
    <div className="login">
      <form>
        <img
          className="logo"
          src="../../../public/XIK_VerdeTransparente.png"
          alt="XikDev"
        />
        <div className="login-inputs">
          <input type="email" placeholder="Email" />
          <input type="password" placeholder="ContraseÃ±a" />
        </div>
        <button>{authConstants.bttnLogin}</button>
      </form>
    </div>
  );
};
