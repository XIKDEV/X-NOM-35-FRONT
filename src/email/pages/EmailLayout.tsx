import '../ui/emailLayout.css';

export const EmailLayout = () => {
  return (
    <div className="email-layout flex-column-center">
      <img src="../../../public/XIK_VerdeTransparente.png" alt="XikDev" />
      <h1>XikDev</h1>

      <div className="email-text flex-column-center">
        <p>
          Bienvenido! <br />
          Por medio de este correo se comparte de manera automática la
          contraseña correspondiente para entrar al sistema de Nom-035. <br />
          Ingresa a [anexar link] o da click al siguiente botón para dirigirte a
          la página e iniciar sesión.
        </p>

        <div className="password flex-column-center">
          <p>Contraseña:</p>
          <p>AKJC64cdsK5!</p>
        </div>

        <a>Nom-035</a>

        <p>
          <span>
            Este mensaje es generado de manera automática, favor de no responder
            directamente este correo, para realizar alguna consulta o tratar
            algún tema, enviar correo a xikdev@gmail.com
          </span>
        </p>
      </div>
    </div>
  );
};
