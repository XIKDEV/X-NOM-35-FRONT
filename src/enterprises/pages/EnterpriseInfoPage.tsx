import { FC } from 'react';
import { Layout } from 'antd';
import { HeaderMobile } from '../../ui/components';
import {
  ContractsList,
  HeadEnterpriseInfo,
  InfoEnterprise,
} from '../components';
import { useScreenSize } from '../../hooks';
import { IEnterpriseInfoPageProps } from '../../interfaces';
import { enterpriseContentStyle } from '../ui';
import '../ui/enterpriseInfoPage.css';

const { Content } = Layout;

export const EnterpriseInfoPage: FC<IEnterpriseInfoPageProps> = ({
  handleSidebar,
}) => {
  const { isMobile } = useScreenSize();

  // //! TODO: temp info
  const enterpriseInfo = {
    enterpriseName: 'XikDev',
    addres: 'Agustín Lara #3251 Col. Miramar, Tijuana, Baja California, México',
    goals: 'Goals',
    activity: 'Industria Tec',
    mainActivity: 'Desarrollo de web',
    quantityEmployees: '2',
    contractsData: [
      {
        contract: 'Nom-contract',
        date: 'mm/aa - mm/aa',
      },
      {
        contract: 'Nom-contract',
        date: 'mm/aa - mm/aa',
      },
      {
        contract: 'Nom-contract',
        date: 'mm/aa - mm/aa',
      },
    ],
  };

  return (
    <>
      {isMobile ? <HeaderMobile handleSidebar={handleSidebar} /> : <></>}
      <HeadEnterpriseInfo enterpriseName={enterpriseInfo.enterpriseName} />
      {isMobile ? (
        <>
          <InfoEnterprise
            addres={enterpriseInfo.addres}
            goals={enterpriseInfo.goals}
            activity={enterpriseInfo.activity}
            mainActivity={enterpriseInfo.mainActivity}
            quantityEmployees={enterpriseInfo.quantityEmployees}
          />
          <ContractsList data={enterpriseInfo.contractsData} />
        </>
      ) : (
        <Content style={enterpriseContentStyle}>
          <InfoEnterprise
            addres={enterpriseInfo.addres}
            goals={enterpriseInfo.goals}
            activity={enterpriseInfo.activity}
            mainActivity={enterpriseInfo.mainActivity}
            quantityEmployees={enterpriseInfo.quantityEmployees}
          />
          <ContractsList data={enterpriseInfo.contractsData} />
        </Content>
      )}
    </>
  );
};
