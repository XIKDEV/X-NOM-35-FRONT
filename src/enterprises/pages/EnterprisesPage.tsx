import { FC } from 'react';
import { EnterprisesCard } from '../components';
import { CardMobile, HeaderPage } from '../../ui/components';
import { useEnterprisesPage } from '../hooks';
import { IEnterprisesPageProps } from '../../interfaces';
import { enterprisesConstants } from '../../constants';
import '../ui/enterprisesPage.css';

export const EnterprisesPage: FC<IEnterprisesPageProps> = ({
  handleSidebar,
}) => {
  const { enterprises, isMobile, contextHolder } = useEnterprisesPage();

  //! TODO: Info temp
  const logoEnterprise = '../../../public/_DSC7606.JPG';

  return (
    //! TODO: Arreglar grid
    <>
      {contextHolder}

      <HeaderPage
        handleSidebar={handleSidebar}
        data-testid="header-desktop"
        formAdd="addEnterprise"
      />
      {isMobile ? (
        <>
          {enterprises.map((enterprise) => (
            <CardMobile
              key={enterprise.id}
              srcImage={logoEnterprise}
              alt={`${enterprisesConstants.logoFrom} ${enterprise.business_name}`}
              title={enterprise.business_name}
              subtitle={`${enterprisesConstants.responsibleEnterprise} ${enterprise.legal_representative}`}
              data-testid="card-mobile"
            />
          ))}
        </>
      ) : (
        <>
          <section className="grid-enterprises-cards">
            {enterprises.map((enterprise) => (
              <EnterprisesCard
                key={enterprise.id}
                srcImage={logoEnterprise}
                alt={`${enterprisesConstants.logoFrom} ${enterprise.business_name}`}
                title={enterprise.business_name}
                subtitle={`${enterprisesConstants.responsibleEnterprise} ${enterprise.legal_representative}`}
                data-testid="card-desktop"
              />
            ))}
          </section>
        </>
      )}
    </>
  );
};
