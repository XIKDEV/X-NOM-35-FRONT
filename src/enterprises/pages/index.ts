export * from './EnterpriseInfoPage';
export * from './EnterprisesPage';
