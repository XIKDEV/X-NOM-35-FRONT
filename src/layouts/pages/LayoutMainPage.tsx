import { Drawer, Layout } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars, faPlus } from '@fortawesome/free-solid-svg-icons';
import { SideBar } from '../components';
import { useLayoutMainPage } from '../hooks';
import { useScreenSize } from '../../hooks';
import { ILayoutMainPageProps } from '../../interfaces';
import { drawerStyles, siderStyles } from '../ui';
import '../ui/layouts.css';

const { Content, Header, Sider } = Layout;

export const LayoutMainPage: React.FC<ILayoutMainPageProps> = ({
  page: Page,
}) => {
  const { isCollapse, handleSidebar } = useLayoutMainPage();
  const { isMobile } = useScreenSize();

  const sideBarProps = {
    isMobile,
    isCollapse,
    handleSidebar,
  };
  const sideBarComponent = <SideBar {...sideBarProps} />;

  return (
    <Layout style={{ maxHeight: '100vh' }}>
      {isMobile ? (
        <Drawer
          open={!isCollapse}
          onClose={handleSidebar}
          closable={false}
          placement={drawerStyles.placement}
          width={drawerStyles.width}
          bodyStyle={drawerStyles.bodyStyle}
          data-testid="drawer"
        >
          {sideBarComponent}
        </Drawer>
      ) : (
        <Sider
          className="flex-column-center space-between"
          style={siderStyles}
          width={186}
          collapsedWidth={64}
          collapsed={isCollapse}
          data-testid="sider"
        >
          {sideBarComponent}
        </Sider>
      )}

      <Content className="content-layout">
        {isMobile ? (
          <Header className="nav flex-row-center space-between">
            <button onClick={handleSidebar} data-testid="sidebar-toggle-button">
              <FontAwesomeIcon icon={faBars} />
            </button>
            <input className="br-4" type="text" placeholder="Búsqueda rápida" />
            <button>
              <FontAwesomeIcon icon={faPlus} />
            </button>
          </Header>
        ) : (
          <></>
        )}
        <Page />
      </Content>
    </Layout>
  );
};
