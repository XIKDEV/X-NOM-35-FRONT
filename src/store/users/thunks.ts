import { getAdapter, sublink } from '../../api';
import { content } from '../../api/routes';

export const getUsersList = () => {
  return async (dispatch: CallableFunction) => {
    import('../request/requestSlice').then(({ setLoading }) => {
      dispatch(setLoading());
    });

    const token = localStorage.getItem('token');
    const header = {
      content,
      auth: `Bearer ${token}`,
    };

    try {
      const { data } = await getAdapter(sublink.user, header);
      import('./usersSlice').then(({ setUsers }) => {
        dispatch(setUsers(data));
      });
      import('../request').then(({ setStopLoading }) => {
        dispatch(setStopLoading());
      });
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
    } catch (err: any) {
      console.clear();
      import('../request').then(({ setErrorInRequest }) => {
        dispatch(setErrorInRequest(err.response.data.message));
      });
    }
  };
};
