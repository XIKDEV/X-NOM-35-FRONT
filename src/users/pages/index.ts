export * from './UserInfoPage';
export * from './UsersPage';
