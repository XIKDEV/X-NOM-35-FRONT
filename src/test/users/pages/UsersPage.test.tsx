import { render } from '@testing-library/react';
import { UsersPage } from '../../../users/pages';

describe('Test in <UserPage />', () => {
  test('should be match with the snapshot', () => {
    const { asFragment } = render(<UsersPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
