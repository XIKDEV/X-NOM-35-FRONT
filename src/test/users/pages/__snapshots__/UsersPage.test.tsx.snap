// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <UserPage /> should be match with the snapshot 1`] = `
<DocumentFragment>
  <div>
    UsersPage
  </div>
</DocumentFragment>
`;
